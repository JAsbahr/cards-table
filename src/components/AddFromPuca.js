import React from "react";
import { connect } from "react-redux"
import CardFormFromPuca from "./CardFormFromPuca"
import { startAddFromPuca } from "../actions/frompuca"
import { startAddFromPucaPromoted } from "../actions/frompucaPromoted"

export class AddFromPuca extends React.Component {
    onSubmit = (card) => {
        this.props.startAddFromPuca(card)
        this.props.history.push("/frompuca")  //history can be used cause of react-router (this component is used in a router)
    }
    onSubmitPromo = (card) => {
        this.props.startAddFromPucaPromoted(card)
        this.props.history.push("/frompucapromoted")
    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">{this.props.location.pathname === "/frompuca/create" ? "Add from" : "Add from promo"}</h1>
                    </div>
                </div>
                <div className="content-container">
                    <CardFormFromPuca
                        onSubmit={this.props.location.pathname === "/frompuca/create" ? this.onSubmit : this.onSubmitPromo} // props.dispatch(addExpense(expense)) code, bevor getestet werden musste
                    />
                </div>
            </div>
        )
    }
}


const mapDispatchtoProps = (dispatch) => ({                  // eingefÃ¼hrt, um leichter zu testen 
    startAddFromPuca: (card) => dispatch(startAddFromPuca(card)),
    startAddFromPucaPromoted: (card) => dispatch(startAddFromPucaPromoted(card))
})

export default connect(undefined, mapDispatchtoProps)(AddFromPuca);
