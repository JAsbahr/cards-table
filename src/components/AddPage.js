import React from "react";
import { connect } from "react-redux"
import ExpenseForm from "./ExpenseForm"
import { addExpense } from "../actions/expenses"

export class AddPage extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense)
        this.props.history.push("/")
    }
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit} // props.dispatch(addExpense(expense)) code, bevor getestet werden musste
                />
            </div>
        )
    }
}


const mapDispatchtoProps = (dispatch) => ({                  // eingefÃ¼hrt, um leichter zu testen 
    addExpense: (expense) => dispatch(addExpense(expense))
})

export default connect(undefined, mapDispatchtoProps)(AddPage);