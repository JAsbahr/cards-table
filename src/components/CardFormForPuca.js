import React from "react"


export default class CardFormForPuca extends React.Component {
    constructor(props) {
        super(props)

        this.state = {
            description: props.card ? props.card.description : "",
            euro: props.card ? (props.card.euro / 100).toString() : "",
            amount: props.card ? (props.card.amount).toString() : "",
            shipping: props.card ? (props.card.shipping / 100).toString() : "",
            sent: props.card ? (props.card.sent).toString() : "",
            puca: props.card ? (props.card.puca).toString() : "",
            error: ""
        }
    }

    onDescriptionChange = (e) => {
        const description = e.target.value
        this.setState(() => ({ description }))
    }
    onAmountChange = (e) => {
        const amount = e.target.value
        if (!amount || amount.match(/^\d{1,}$/)) {
            this.setState(() => ({ amount }))
        }
    }
    onEuroChange = (e) => {
        const euro = e.target.value
        if (!euro || euro.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ euro }))
        }
    }
    onShippingChange = (e) => {
        const shipping = e.target.value
        if (!shipping || shipping.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ shipping }))
        }
    }
    onSentChange = (e) => {
        const sent = e.target.value
        if (!sent || sent.match(/^\d{1,}$/)) {
            this.setState(() => ({ sent }))
        }
    }
    onPucaChange = (e) => {
        const puca = e.target.value
        if (!puca || puca.match(/^\d{1,}$/)) {
            this.setState(() => ({ puca }))
        }
    }
    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.description || !this.state.amount || !this.state.amount) {
            this.setState(() => ({
                error: "Please provide card name, amount and cost"
            }))
        } else {
            this.setState(() => ({ error: "" }))
            this.props.onSubmit({
                description: this.state.description,
                euro: parseFloat(this.state.euro) * 100,
                amount: parseFloat(this.state.amount),
                shipping: this.state.shipping === "" ? 0 : parseFloat(this.state.shipping) * 100,
                sent: this.state.sent === "" ? 0 : parseFloat(this.state.sent),
                puca: this.state.puca === "" ? 0 : parseFloat(this.state.puca)
            })
        }
    }

    render() {
        return (
            <form className="form" onSubmit={this.onSubmit}>
                {this.state.error && <p className="form__error">{this.state.error}</p>}
                <input
                    type="text"
                    placeholder="Card name"
                    autoFocus
                    className="text-input"
                    value={this.state.description}
                    onChange={this.onDescriptionChange}
                />
                <input
                    type="text"
                    className="text-input"
                    placeholder="Amount"
                    value={this.state.amount}
                    onChange={this.onAmountChange}
                />
                <input
                    type="text"
                    className="text-input"
                    placeholder="Cost in Euro"
                    value={this.state.euro}
                    onChange={this.onEuroChange}
                />
                <input
                    type="text"
                    className="text-input"
                    placeholder="Amount sent"
                    value={this.state.sent}
                    onChange={this.onSentChange}
                />
                <input
                    type="text"
                    className="text-input"
                    placeholder="Shipping costs"
                    value={this.state.shipping}
                    onChange={this.onShippingChange}
                />
                <input
                    type="text"
                    className="text-input"
                    placeholder="Puca value"
                    value={this.state.puca}
                    onChange={this.onPucaChange}
                />
                <div>
                    <button className="button">Save card</button>
                </div>
            </form>
        )
    }
}
