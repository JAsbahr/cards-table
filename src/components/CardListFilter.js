import React from "react"
import { connect } from "react-redux"
import { setTextFilter } from "../actions/filters"
import { Link } from "react-router-dom"


export class CardListFilter extends React.Component {
    constructor(props) {
        super(props)
    }

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }

    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            className="text-input"
                            type="text"
                            value={this.props.filters.text}
                            onChange={this.onTextChange}
                            placeholder="Search for card"
                        />
                    </div>
                    <div className="input-group__item">
                        <Link className="button button--green" to="frompuca/create">Add from</Link>
                    </div>
                    <div className="input-group__item">
                        <Link className="button button--green" to="forpuca/create">Add for</Link>
                    </div>
                    <div className="input-group__item">
                        <Link className="button button--green" to="frompucapromoted/create">Add promo from</Link>
                    </div>
                    <div className="input-group__item">
                        <Link className="button button--green" to="forpucapromoted/create">Add promo for</Link>
                    </div>
                </div>
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text))
})

export default connect(mapStateToProps, mapDispatchToProps)(CardListFilter)
