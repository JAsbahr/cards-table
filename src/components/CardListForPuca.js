import React from "react"
import { connect } from "react-redux"
import CardListItemForPuca from "./CardListItemForPuca"
import getVisibleCards from "../selectors/getVisibleCards"

export const CardListForPuca = (props) => { // export for testing
    const arrayLength = 7

    return (
        <div className="content-container">
            <div className="list-header">
                <div className="list-item-long">Card name</div>
                <div className="list-item-short">Amount</div>
                <div className="list-item-short">Cost</div>
                <div className="list-item-very-short">Sent</div>
                <div className="list-item-middle">Shipping costs</div>
                <div className="list-item-short">Puca value</div>
                <div className="list-item-short">Percentage</div>
            </div>
            {props.location === "/forpuca" ? <div className="list-body">
                {
                    props.cards.length === 0 ? (
                        <div className="list-item list-item--message">
                            <span>No cards found</span>
                        </div>
                    ) : (
                            props.cards.length > 7 ? 
                            props.cards.slice(props.cards.length - arrayLength, props.cards.length).reverse().map((card) => {
                                return <CardListItemForPuca key={card.id} location={props.location} {...card} />
                            }) :
                            props.cards.reverse().map((card) => {
                                return <CardListItemForPuca key={card.id} location={props.location} {...card} />
                            })
                        )
                }
            </div> : <div className="list-body">
                    {
                        props.cardsPromoted.length === 0 ? (
                            <div className="list-item list-item--message">
                                <span>No cards found</span>
                            </div>
                        ) : (
                                props.cardsPromoted.length > 7 ?
                                props.cardsPromoted.slice(props.cardsPromoted.length - arrayLength, props.cardsPromoted.length).reverse().map((card) => {
                                    return <CardListItemForPuca key={card.id} location={props.location} {...card} />
                                }) : 
                                props.cardsPromoted.reverse().map((card) => {
                                    return <CardListItemForPuca key={card.id} location={props.location} {...card} />
                                })
                            )
                    }
                </div>}
        </div>
    )
}

const mapStatetoProps = (state) => {
    return {
        cards: getVisibleCards(state.forpuca, state.filters),
        cardsPromoted: getVisibleCards(state.forpucaPromoted, state.filters)
    }
}

export default connect(mapStatetoProps)(CardListForPuca);  // HOC (kind of)
