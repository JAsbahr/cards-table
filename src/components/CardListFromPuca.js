import React from "react"
import { connect } from "react-redux"
import CardListItemFromPuca from "./CardListItemFromPuca"
import getVisibleCards from "../selectors/getVisibleCards"

export const CardListFromPuca = (props) => { // export for testing
    const arrayLength = 7

    return (
        <div className="content-container">
            <div className="list-header">
                <div className="list-item-long">Card name</div>
                <div className="list-item-short">Amount</div>
                <div className="list-item-short">Puca value</div>
                <div className="list-item-very-short">Sent</div>
                <div className="list-item-short">Sold for</div>
                <div className="list-item-middle"></div>
                <div className="list-item-short">Percentage</div>
            </div>
            {props.location === "/frompuca" ? <div className="list-body">
                {
                    props.cards.length === 0 ? (
                        <div className="list-item list-item--message">
                            <span>No cards found</span>
                        </div>
                    ) : (
                            props.cards.slice(props.cards.length - arrayLength, props.cards.length).reverse().map((card) => {
                                return <CardListItemFromPuca key={card.id} location={props.location} {...card} />
                            })
                        )
                }
            </div> : <div className="list-body">
                    {
                        props.cardsPromoted.length === 0 ? (
                            <div className="list-item list-item--message">
                                <span>No cards found</span>
                            </div>
                        ) : (
                                props.cardsPromoted.slice(props.cardsPromoted.length - arrayLength, props.cardsPromoted.length).reverse().map((card) => {
                                    return <CardListItemFromPuca key={card.id} location={props.location} {...card} />
                                })
                            )
                    }
                </div>}
        </div>
    )
}

const mapStatetoProps = (state) => {
    return {
        cards: getVisibleCards(state.frompuca, state.filters),
        cardsPromoted: getVisibleCards(state.frompucaPromoted, state.filters)
    }
}

export default connect(mapStatetoProps)(CardListFromPuca);  // HOC (kind of)
