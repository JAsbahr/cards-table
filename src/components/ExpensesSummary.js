import React from "react"
import { connect } from "react-redux"
import getVisibleExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/selectExpensesTotal";
import numeral from "numeral"

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? "expense" : "expenses"
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div>
            <p>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</p>
        </div>
    )
}

const mapStatetoProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStatetoProps)(ExpensesSummary)