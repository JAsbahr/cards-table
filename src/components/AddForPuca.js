import React from "react";
import { connect } from "react-redux"
import CardFormForPuca from "./CardFormForPuca"
import { startAddForPuca } from "../actions/forpuca"
import { startAddForPucaPromoted } from "../actions/forpucaPromoted"

export class AddForPuca extends React.Component {
    onSubmit = (card) => {
        this.props.startAddForPuca(card)
        this.props.history.push("/forpuca")  //history can be used cause of react-router (this component is used in a router)
    }
    onSubmitPromo = (card) => {
        this.props.startAddForPucaPromoted(card)
        this.props.history.push("/forpucapromoted")
    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">{this.props.location.pathname === "/forpuca/create" ? "Add for" : "Add for promo"}</h1>
                    </div>
                </div>
                <div className="content-container">
                    <CardFormForPuca
                        onSubmit={this.props.location.pathname === "/forpuca/create" ? this.onSubmit : this.onSubmitPromo}
                    />
                </div>
            </div>
        )
    }
}


const mapDispatchtoProps = (dispatch) => ({                  // eingefÃ¼hrt, um leichter zu testen 
    startAddForPuca: (card) => dispatch(startAddForPuca(card)),
    startAddForPucaPromoted: (card) => dispatch(startAddForPucaPromoted(card))
})

export default connect(undefined, mapDispatchtoProps)(AddForPuca);
