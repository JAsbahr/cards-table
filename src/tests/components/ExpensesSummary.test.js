import React from "react"
import { shallow } from "enzyme"
import { ExpensesSummary } from "../../components/ExpensesSummary"

test("should render the ExpensesSummary correctly with 1 expense", () => {
    const result = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>)
    expect(result).toMatchSnapshot()
})

test("should render the ExpensesSummary correctly with more than 1 expense", () => {
    const result = shallow(<ExpensesSummary expenseCount={23} expensesTotal={15324}/>)
    expect(result).toMatchSnapshot()
})