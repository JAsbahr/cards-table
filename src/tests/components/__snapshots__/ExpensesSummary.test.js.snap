// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </p>
</div>
`;

exports[`should render the ExpensesSummary correctly with more than 1 expense 1`] = `
<div>
  <p>
    Viewing 
    23
     
    expenses
     totalling 
    $153.24
  </p>
</div>
`;
